black.py

```diff
@@ -138,7 +150,10 @@ def main(ctx: click.Context, line_length: int, fast: bool, src: List[str]) -> No
 def format_file_in_place(src: Path, line_length: int, fast: bool) -> bool:
     """Format the file and rewrite if changed. Return True if changed."""
     try:
-        contents, encoding = format_file(src, line_length=line_length, fast=fast)
+        with tokenize.open(src) as src_buffer:
```

Comment:
Note: with my suggested changes to `main()`, if a user passes more than one file name, the tool will fail with "FileNotFoundError: [Errno 2] No such file or directory: '-'" and I think this is clear enough of an indication that it wasn't treated as stdin by the tool then. Since the report at the end will show other files being properly handled, this should be enough of a nudge to the user that `-` is special.